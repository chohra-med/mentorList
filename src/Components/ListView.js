import React, {Component} from 'react';
import {
    View,
    ScrollView,
    Image,
    TouchableOpacity,
    TextInput,
    Text,
    RefreshControl,
    Switch,
    I18nManager,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

//we import Loadach to filter Items
import _ from 'lodash';
import I18n from '../locales/i18n';
import {PROFILE} from "../assets/Images";


// Data that will be showed first
const firstData = [
    {"id": 1, "helps": "Software engineer", "name": "Chohra Mohammed"},
    {"id": 2, "helps": "Junior Mobile Developer", "name": "Amir Khoutir"},
    {"id": 3, "helps": "Junior Data Science", "name": "Chellaf Younes"},
    {"id": 4, "helps": "Senior Data Science", "name": "Bederrer Alladin"},
    {"id": 5, "helps": "Junior Mobile Developer", "name": "Benyamina Mouad"},
    {"id": 6, "helps": "Senior FullStack Developer", "name": "Sabour Oussama"},
    {"id": 7, "helps": "Lead Fronted Developer", "name": "Oumellal Abdeslam"},
    {"id": 8, "helps": "Senior Backend Developer", "name": "Bouregag Abdelkaded"},
    {"id": 9, "helps": "Senior Supply chain Specialist", "name": "Borsali Faycel"},
    {"id": 10, "helps": "Junior Digital Marketing", "name": "Bouguesri Adel"}
];

// Data that will be showed after
const secondData = [
    {"id": 412, "helps": "Junior Devops Developer", "name": "Neddar Islem"},
    {"id": 323, "helps": "Senior Blockchain Developer", "name": "Khodja Mehdi"},
    {"id": 512, "helps": "Senior Blockchain Developer", "name": "Bacha Brahim"},
    {"id": 47, "helps": "Junior Web Developer", "name": "Damene Miloud "},
    {"id": 58, "helps": "CEO", "name": "Touaria Sami"},
    {"id": 69, "helps": "Business developer", "name": "Hafsawi Houssem"},
    {"id": 610, "helps": "Junior Ui/Ux Designer", "name": "Kafi Abderrazak"}
];

//this function return if ScrollView reached the buttom or not
const isCloseToBottom = ({layoutMeasurement, contentOffset, contentSize}) => {
    const paddingToBottom = 20;
    return layoutMeasurement.height + contentOffset.y >=
        contentSize.height - paddingToBottom;
};

export default class ListView extends Component {
    constructor(props) {
        super(props);
        this.state = {
            loadMore: false,
            ourData: [],
            generalData: [],
            nameFiltred: '',
            refreshing: false,
            isRTL: true,
            currentLanguage: 'en'
        };
        this.filterby = this.filterby.bind(this);
        this.orderby = this.orderby.bind(this);
    };

    //to filter our data by NAME
    filterby() {
        let data = this.state.generalData;
        data = data.filter(x => String(x.name.toUpperCase()).includes(this.state.nameFiltred.toUpperCase()));
        this.setState({ourData: data})
    }

    //to order our data by NAME
    orderby() {
        let data = this.state.ourData;
        data = _.sortBy(data, ['name', 'id']);
        this.setState({ourData: data})
    }

    //We recupere our data and put it in our State, And we set the language
    componentWillMount() {
        this.setState({ourData: firstData});
        this.setState({generalData: firstData});
        this.setState({isRTL: !I18nManager.isRTL});
        if (I18nManager.isRTL) {
            this.setState({currentLanguage: 'ar'});
        } else {
            this.setState({currentLanguage: 'en'});

        }
    }

    // this is called when we want to add the second data
    // When scrollView reached its end
    updateList() {
        if (!this.state.loadMore) {
            let data = this.state.ourData;
            data = data.concat(secondData);
            this.setState({generalData: data});
            this.setState({ourData: data});
            this.setState({loadMore: true});
            this.setState({nameFiltred: ''})
        }
    }

    //This enable us to control the direction for different language
    _onDirectionChange = () => {
        I18nManager.forceRTL(this.state.isRTL);
        this.setState({isRTL: !this.state.isRTL});
        if (this.state.isRTL) {
            this.setState({currentLanguage: 'ar'});
        } else {
            this.setState({currentLanguage: 'en'});
        }
    };

//This function enable us to MAP our Data in the our form
    showList() {
        return (
            this.state.ourData.map(data => {
                return (
                    <TouchableOpacity key={data.id}>
                        <View style={styles.listViewContainer}>
                            <Image
                                style={styles.image}
                                source={PROFILE}
                            >
                            </Image>
                            <View style={styles.textContainer}>
                                <Text style={styles.name}>{data.name}</Text>
                                <Text style={styles.helps}> {data.helps}</Text>

                            </View>
                            <Icon name="arrow-right" style={styles.button}
                                  color='black'
                                  size={32}
                                  onPress={() => {
                                  }}
                            >
                            </Icon>
                        </View>
                        <View
                            style={{
                                borderBottomColor: 'black',
                                borderBottomWidth: 0.5,
                                width: '90%',
                                marginLeft: '5%'
                            }}
                        />
                    </TouchableOpacity>
                )
            })
        );
    }

    //this is called when we refresh our scroll view
    _onRefresh = () => {
        this.setState({refreshing: true});
        this.setState({ourData: this.state.generalData});
        this.setState({refreshing: false});
        this.setState({nameFiltred: ''})
    };


    render() {
        let {currentLanguage, nameFiltred, refreshing} = this.state;
        I18n.locale = currentLanguage;
        I18n.fallbacks = true;
        return (
            <View>
                <View style={{
                    flexDirection: 'row',
                    justifyContent: 'center'
                }}>
                    <TextInput
                        style={{height: 40, borderWidth: 2, margin: 5, width: '60%'}}
                        placeholder={I18n.t('filtering')}
                        value={nameFiltred}
                        onChangeText={(nameFiltred) => {
                            this.setState({nameFiltred});
                            this.filterby;
                        }}
                    />
                    <TouchableOpacity
                        style={{width: 30, justifyContent: 'center', borderRadius: 20, borderColor: 'grey', margin: 6}}
                        onPress={this.filterby}
                    >
                        <Icon color='black' size={20} name="search"/>
                    </TouchableOpacity>

                    <TouchableOpacity
                        style={{
                            justifyContent: 'center',
                            borderRadius: 4,
                            borderColor: 'grey',
                            borderWidth: 1,
                            marginLeft: 20,
                            margin: 6
                        }}
                        onPress={this.orderby}
                    >
                        <Text style={{fontSize: 14, color: 'black'}}> {I18n.t('sortElement')}</Text>
                    </TouchableOpacity>

                </View>
                <View title={'Quickly Test RTL Layout'}>
                    <View style={{flexDirection: 'row', justifyContent: 'center'}}>
                        <Text style={[styles.name, {margin: 1}]}> {I18n.t('changeLanguage')}</Text>
                        <View style={styles.switchRowSwitchView}>
                            <Switch
                                onValueChange={this._onDirectionChange}
                                style={styles.rightAlignStyle}
                                value={this.state.isRTL}
                            />

                        </View>
                    </View>
                </View>
                <ScrollView

                    refreshControl={
                        <RefreshControl
                            refreshing={refreshing}
                            onRefresh={this._onRefresh}
                        />}

                    contentContainerStyle={{flexGrow: 1, borderWidth: 1, paddingBottom: 80}}

                    onScroll={({nativeEvent}) => {
                        if (isCloseToBottom(nativeEvent)) {
                            this.updateList()
                        }
                    }}
                >
                    <View>
                        {
                            this.showList()
                        }
                    </View>
                </ScrollView>
            </View>
        );
    }
}
