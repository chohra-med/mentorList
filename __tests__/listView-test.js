import ListView from "../src/Components/ListView";
import React from "react";

import renderer from "react-test-renderer";


test('renders the ListView', () => {
    const listView = renderer.create(<ListView/>).toJSON();
    expect(listView).toMatchSnapshot();
});

test('Test First Data', () => {
    const listView = renderer.create(<ListView/>).getInstance();
    let firstData = listView.state.ourData;
    expect(firstData.length).toEqual(10);
});
test('Test Loading More', () => {
    const listView = renderer.create(<ListView/>).getInstance();
    let data = listView.state.ourData;
    listView.orderBy();
    let dataAfter = listView.state.ourData;
    let equal = JSON.stringify(dataAfter) === JSON.stringify(data);
    expect(equal).toBeFalsy();
});

describe('Sorting', () => {
    it('Array should be sortedby name , id', () => {
        const listView = renderer.create(<ListView/>).getInstance();
        let data = listView.state.ourData;
        listView.orderBy();
        let dataAfter = listView.state.ourData;

        data.sort((a, b) => {
            if (a.name > b.name) {
                return 1;
            }
            if (b.name > a.name) {
                return -1;
            }
            else {
                return a.id-b.id;
            }
        });
        let equal = JSON.stringify(dataAfter) === JSON.stringify(data);
        expect(equal).toBeTruthy();
    });
});

