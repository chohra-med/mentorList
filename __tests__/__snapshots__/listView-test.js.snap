// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ListView 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Type here to filter by NAME"
      rejectResponderTermination={true}
      style={
        Object {
          "borderWidth": 2,
          "height": 40,
          "margin": 5,
          "width": "60%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "grey",
          "borderRadius": 20,
          "justifyContent": "center",
          "margin": 6,
          "opacity": 1,
          "width": 30,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "grey",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "margin": 6,
          "marginLeft": 20,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 14,
          }
        }
      >
         
        SortElements
      </Text>
    </View>
  </View>
  <View
    title="Quickly Test RTL Layout"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 16,
            },
            Object {
              "margin": 1,
            },
          ]
        }
      >
         
        change Language
      </Text>
      <View>
        <RCTSwitch
          accessibilityRole="button"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 31,
              "width": 51,
            }
          }
          value={true}
        />
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "borderWidth": 1,
        "flexGrow": 1,
        "paddingBottom": 80,
      }
    }
    onScroll={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <RCTRefreshControl />
    <View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Chohra Mohammed
              </Text>
              <Text
                style={Object {}}
              >
                 
                Software engineer
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Amir Khoutir
              </Text>
              <Text
                style={Object {}}
              >
                 
                Junior Mobile Developer
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Chellaf Younes
              </Text>
              <Text
                style={Object {}}
              >
                 
                Junior Data Science
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Bederrer Alladin
              </Text>
              <Text
                style={Object {}}
              >
                 
                Senior Data Science
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Benyamina Mouad
              </Text>
              <Text
                style={Object {}}
              >
                 
                Junior Mobile Developer
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Sabour Oussama
              </Text>
              <Text
                style={Object {}}
              >
                 
                Senior FullStack Developer
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Oumellal Abdeslam
              </Text>
              <Text
                style={Object {}}
              >
                 
                Lead Fronted Developer
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Bouregag Abdelkaded
              </Text>
              <Text
                style={Object {}}
              >
                 
                Senior Backend Developer
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Borsali Faycel
              </Text>
              <Text
                style={Object {}}
              >
                 
                Senior Supply chain Specialist
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 80,
                "justifyContent": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/profile.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 60,
                  "height": 60,
                  "left": "5%",
                  "margin": 10,
                  "position": "absolute",
                  "width": 60,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "margin": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  }
                }
              >
                Bouguesri Adel
              </Text>
              <Text
                style={Object {}}
              >
                 
                Junior Digital Marketing
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 32,
                  },
                  Object {
                    "marginTop": "5%",
                    "position": "absolute",
                    "right": "5%",
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 0.5,
                "marginLeft": "5%",
                "width": "90%",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
